relay-server/src/actors/global_config.rs

```diff
@@ -310,14 +334,69 @@ mod tests {
         });
 
         Controller::start(Duration::from_secs(1));
-        let config = Arc::<Config>::default();
-        let service = GlobalConfigService::new(config.clone(), upstream).start();
+        let mut config = Config::default();
+        config.update_credentials();
+        let fetch_interval = config.global_config_fetch_interval();
+
+        let service = GlobalConfigService::new(Arc::new(config), upstream).start();
 
         assert!(service.send(Get).await.is_ok());
 
         Controller::shutdown(ShutdownMode::Immediate);
-        tokio::time::sleep(config.global_config_fetch_interval() * 2).await;
+        tokio::time::sleep(fetch_interval * 2).await;
 
         assert!(service.send(Get).await.is_ok());
     }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn managed_relay_makes_upstream_request() {
```

Comment:
nice job fixing this! was it the timer that you added which fixed it ?

Reply:
I don't remember what the error was, but the missing piece initially was the `#[should_panic]` if that's what you mean.

Reply:
I think I see the problem, when I wanted to return a bool from the mock, itd wait until the response from upstream and thereby always fail as an error would be logged. Pity because it'd make it easier to write these tests and would make them more explicit.